pkg/controller/operators/catalog/operator.go

```diff
@@ -994,7 +994,7 @@ func (o *Operator) syncRegistryServer(logger *logrus.Entry, in *v1alpha1.Catalog
 	}
 
 	healthy, err := srcReconciler.CheckRegistryServer(logger, in)
-	if err != nil {
+	if err != nil && !healthy {
```

Comment:
@joelanford had add in this change to make sure we're sending the reconciler back if there were errors force deleting the dead pods, AND `CheckRegistryServer` returns `healthy=false`. In case we detect dead pods but run into errors deleting them, but we find healthy pod, we are going forward with the current loop.

Reply:
If we return `healthy=true` from `CheckRegistryServer`, but there was a problem deleting some dead pod, isn't the whole point of returning `healthy=true` moot?

Reply:
If we encounter an error doing stuff, we should bubble that error up regardless. My original point is that we should be able to return `healthy=true, err=nil` when there is a live pod and we successfully delete the dead pods.

Pretend like we don't own the implementation of `CheckRegistryServer`. If we add `!healthy` to this check, then we'll completely ignore errors that occur during `CheckRegistryServer` if it returns `healthy=true`.

If any error is returned we _should_ propagate that error even if there are other healthy pods, because those errors might be indicative of a separate problem that needs a user to intervene.